crates/sargon/src/signing/petition_types/petition_for_entity.rs

```diff
@@ -5,26 +5,26 @@ use crate::prelude::*;
 /// `{ threshold: PetitionForFactors, override: PetitionForFactors }`
 #[derive(Clone, PartialEq, Eq, derive_more::Debug)]
 #[debug("{}", self.debug_str())]
-pub(crate) struct PetitionForEntity {
+pub(crate) struct PetitionForEntity<S: Signable> {
```

Comment:
Looks like this can be made generic over `T: SignableID` only, not the whole `Signable` trait?